testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/SAMLServletAdapterTest.java

```diff
@@ -804,8 +804,7 @@ public void salesPostTestCompositeRoleForUser() {
         UserRepresentation topGroupUser = createUserRepresentation("topGroupUser", "top@redhat.com", "", "", true);
         setPasswordFor(topGroupUser, PASSWORD);
 
-        String expectedString = StoreProvider.getCurrentProvider().isMapStore() ? "principal=topGroupUser" : "principal=topgroupuser";
```

Comment:
I'd like to see a code comment describing the reason for having this switch, for example "map storage preserves the case of users" - otherwise it would be hard for future readers to understand why the statement is there. Alternatively, you could have a case-insensitive comparison for the suffix (everything after "principal=").

Looking at it again, and with the changes in the mapper, this isn't something that depends on the map store, but on the realm setting in a map store. With the default being false (as now implemented in the mapper), it should be lower case.

Reply:
So do I get it right there is no need to add the comment there?

Reply:
Yes - when the change is reverted, no comment is needed.